src/presentation/interfaces/http-response-formatter.ts

```diff
@@ -0,0 +1,7 @@
+import { ErrorCode } from '../../model/enums/error-code'
+
+export interface HttpResponseFormatter {
+  formatSuccessResponse(data?: object): object
+  formatFailResponse(data?: object): object
+  formatErrorResponse(message: string, errorCode: ErrorCode): object
```

Comment:
Since this is most likely going to be mostly used with Exceptions it would be nice to have an overload for this method that took our `Exception` class as an argument.